packages/playground-ui/src/components/assistant-ui/tools/tool-fallback.tsx

```diff
@@ -25,15 +25,17 @@ const ToolFallbackInner = ({ toolName, result, args, metadata, ...props }: ToolF
   // The response from the fetch request resolving the workflow might theoretically
   // be resolved after we receive the first stream event
 
-  useWorkflowStream(result);
-  const { data: workflow, isLoading } = useWorkflow(toolName);
+  const isAgent = (metadata?.mode === 'network' && metadata.from === 'AGENT') || toolName.startsWith('agent-');
+  const isWorkflow = (metadata?.mode === 'network' && metadata.from === 'WORKFLOW') || toolName.startsWith('workflow-');
 
-  const isAgent = metadata?.mode === 'network' && metadata.from === 'AGENT';
+  const agentToolName = toolName.startsWith('agent-') ? toolName.substring('agent-'.length) : toolName;
+  const workflowToolName = toolName.startsWith('workflow-') ? toolName.substring('workflow-'.length) : toolName;
 
-  if (isAgent) {
-    const messages = result?.childMessages ?? [];
+  useWorkflowStream(result);
+  const { data: workflow, isLoading } = useWorkflow(workflowToolName, isWorkflow);
```

Comment:
instead of having useWorkflow here, we should probably modify the WorkflowBadge and handle the fetching request instead. It should be clearer.

The stream accumulation in useWorkflowStream should stay there though